js-core/homeworks/homework-3/src/main.js

```diff
@@ -0,0 +1,245 @@
+будет свойство name
+и возвращать новый объект изменяя свойство name
+в большой регистр
+*/
+
+function toUpperCase(obj) {
+	let newObj = {};
+
+	newObj.name = obj.name.toUpperCase();
+
+	return newObj;
+}
+
+const objectWithName = { name: 'privet kak dela' };
+console.log('task 2', toUpperCase(objectWithName));
+
+/* TASK 3-----------------------------------------------------
+1. Напишите функцию которая принимает в
+качестве аргумента массив и элемент массива
+и добавляет элемент в конец массива
+*/
+
+let arr = [];
+arr[0] = 'privet';
+arr[1] = 'poka';
+arr[2] = 'zdrastvuite';
+
+function addToArray(arrParam, newElem) {
+  // без "естественных" способов добавления
+  let arrLength = arrParam.length;
+  //console.log(arrLength);
+
+  arrParam[arrLength] = newElem;
+}
+
+addToArray(arr, 34);
+addToArray(arr, 55);
+console.log('task 3', arr);
+
+/* TASK 4--------------------------------------------------------
+2. Напишите функцию которая получает 3 параметра
+и возвращает объект типа:
+    {
+      argument1: param1,
+      argument2: param2,
+      argument3: param3,
+    }
+*/
+
+/* {
+ argument1:'77',
+ argument2:[{ name: 'Egor' }, { name: 'Katya' }, { name: 'Vera' }],
+ argument3:PRIVET KAK DELA CHTO NOVOGO
+ }*/
+
+function simpleObjectGenerator(one, two, three) {
+	let obj = {};
+	obj.argument1 = one;
+	obj.argument2 = two;
+	obj.argument3 = three;
+	
+	return obj;
+}
+var userNames = [{ name: 'Egor' }, { name: 'Katya' }, { name: 'Vera' }];
+
+console.log('task 4 - 1', simpleObjectGenerator('protocol', { url: '22' }, 8000));
+console.log('task 4 - 2', simpleObjectGenerator(77, userNames, 'privet kak dela chto novogo'.toUpperCase()));
+
+
+/*TASK 5-------------------------------------------------------------
+3.  Напишите функцию которая принимает 3 аргумента,
+     третий аргумент - это объект.
+    Функция создает новый объект и добавляет ключ
+    это первый аргумент, а значение - второй аргумент
+    Проверяет если есть свойство name в переданном объекте,
+    тогда добавляет данное свойство и возвращает новый объект
+*/
+var myName = { name: 'Oleg' };
+
+function addNameToUser(newKey, newValue, userName) {
+	let obj = {};
+
+	obj.newKey = newValue;
+
+	if (userName.name) {
+		obj.name = userName.name;
+		return obj;
+	} else {
+		return 'Sorry, there is no "name" in object';
+	}
+
+}
+
+console.log('task 5 - 1', addNameToUser('family', '%Lustenko%', myName));
+console.log('task 5 - 2', addNameToUser('name', 'privet Pasha', {}));
+/* {name:'Oleg', family:'%Lustenko%'} */
+
+//console.log(myName);
+/* {name:'Oleg'} */
+
+
+/* @@ SUPER !----------------------------------------------------------------
+  Напишите функцию, которая будет
+   возвращать 'Fizz' если передаваемый параметр кратен 3,
+   'Buzz', если передаваемый параметр кратен 5,
+   'FizzBuzz' - если параметер кратен 3 и 5
+   Если передаваемое число не кратно 3 или 5, то вернуть указанный параметр
+*/
+//the first solution----------------------------------
```

Comment:
you have a lot of cool solution. 

But you have to return value from fizzBuzz function, and then you can alert or log it, or something else

Reply:
When you do that way you could think about simplifying your conditionals

Reply:
Do you mean something like that? See the second solution, please

Reply:
Unfortunatelly, I usually have problems with simplifying my code (

Reply:
@viskosa don't worry I could help you with simplifying, just DM me if you have some questions